source/common/filter/config_discovery_impl.h

```diff
@@ -156,13 +162,16 @@ class DynamicFilterConfigProviderImpl : public DynamicFilterConfigProviderImplBa
   // it. Filter factories may hold their own thread local storage which is required to be deleted
   // on the main thread.
   struct MainConfig {
+    MainConfig(ThreadLocal::SlotAllocator& tls)
+        : tls_(std::make_unique<ThreadLocal::TypedSlot<ThreadLocalConfig>>(tls)) {
+      tls_->set([](Event::Dispatcher&) { return std::make_shared<ThreadLocalConfig>(); });
+    }
     absl::optional<FactoryCb> current_config_{absl::nullopt};
+    ThreadLocal::TypedSlotPtr<ThreadLocalConfig> tls_;
```

Comment:
Does this imply that all worker threads will have a pointer to the main-thread's TLS?

Reply:
I don't think so - main config is only passed to the completion callback in `runOnAllThreads` which always executes on the main.